pages/fetch.tsx

```diff
@@ -6,11 +6,11 @@ import { useAppContext } from "../context";
 import style from "../styles/Page.module.css";
 
 export default function Fetch() {
-  const { fetchedDogs } = useAppContext();
-  const numberOfDogs = fetchedDogs.length;
+  const { dogs } = useAppContext();
+  const numberOfDogs = dogs.length;
   let titleOfPage = `🐕 You fetched ${numberOfDogs} dogs!`;
 
-  if (fetchedDogs.length === 0) {
+  if (dogs.length === 0) {
```

Comment:
If you've got the loading state, I'd use that. You can have the case when loading has finished and there is 0 results and you will be displaying user that it's still loading (which is not true).